test/compilable/cppmangle.d

```diff
@@ -976,3 +976,38 @@ else version(Posix)
     static assert(test19278_4.mangleof == "_ZN5hello5world3yay11test19278_4Ev");
     static assert(test19278_var.mangleof == "_ZN5hello5world13test19278_varE");
 }
+
+/*****************************************/
+// https://issues.dlang.org/show_bug.cgi?id=18716
+// D slices as parameters in extern(C++) functions
+
+struct Test18716 {}
+struct __dslice(T) {}
```

Comment:
The tests needs to be in runnable

Reply:
The ABI doesn't match, so there's no point, at least not yet.

Reply:
The point is exactly that the ABI does not match.

Reply:
I'm not following. This whole PR is moot as long as the ABI problem exists.

Reply:
Yep, I was just saying that it's not obvious, since there are tests, and the PR is green.
But IMO, this PR should ultimately be accepted, but it looks like you fear it to be vetoed.

Reply:
Yes. It should, but Walter hasn't responded to the reasoning why his concerns are unfounded.

Reply:
I’m not going to spend time on moving the tests if/until I fix the ABI issue.

Reply:
Walter did already vetoed it.

Reply:
But not sensibly.

Reply:
#11073 contains a `runnable_cxx` test (if someone wants to use it)